### PR TITLE
iOS Push Notifications

### PR DESCRIPTION
**Pull request by masterrr**
Closing https://github.com/toggl/mobile/issues/776 https://github.com/toggl/mobile/issues/775

### CODE REVIEW

**Comment by monday8am** [comment]
@masterrr please, refactor the method name 

https://github.com/toggl/mobile/blob/master/Phoebe/Net/PushRestClient.cs#L60
https://github.com/toggl/mobile/blob/master/Phoebe/Net/PushRestClient.cs#L77

with the Async termination.

File: Ross/AppDelegate.cs [hunk 1]
```diff
@@ -26,6 +26,7 @@ public partial class AppDelegate : UIApplicationDelegate, IPlatformInfo
         private TogglWindow window;
         private int systemVersion;
         private const int minVersionWidget = 7;
+        private bool componentsInitialized = false;
```
Comment (monday8am) [inline Ross/AppDelegate.cs]: we're not using this.

File: Ross/AppDelegate.cs [hunk 2]
```diff
@@ -53,10 +54,27 @@ public override bool FinishedLaunching (UIApplication application, NSDictionary
             ServiceContainer.Resolve<IBugsnagClient> ();
             ServiceContainer.Resolve<BugsnagUserManager> ();
             ServiceContainer.Resolve<ITracker> ();
+            ServiceContainer.Resolve<APNSManager> ();
```
Comment (monday8am) [inline Ross/AppDelegate.cs]: if you are resolving this object there, save the reference and you don't have to resolve it in the event handlers.. (RegisteredForRemoteNotifications, FailedToRegisterForRemoteNotifications, etc)

File: Ross/Net/APNSManager.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+{
+    public class APNSManager
+    {
+        private static readonly string Tag = "APNSManager";
+
+        private const string PushDeviceTokenKey = "PushDeviceToken";
+        private readonly object subscriptionAuthChanged;
+
+        private string authToken;
+        private string SavedDeviceToken { get { return NSUserDefaults.StandardUserDefaults.StringForKey (PushDeviceTokenKey); } }
```
Comment (monday8am) [inline Ross/Net/APNSManager.cs]: use the usual property format.

**Comment by monday8am** [comment]
to new little tasks:
- Check if is needed to do RegisterDevice and UnregisterDevice "awaitables" . Maybe you need to exec one and wait to exec the other one. In case of not, avoid to call them consecutively with the same token.
- Makes DidReceiveRemoteNotification "awaitable" (return a Task) and use try-catch like in Android version. Then await for it in the event handler (DidReceiveRemoteNotification).

the rest for me is ok :+1:

**Comment by masterrr** [comment]
@monday8am check it out now :+1:

**Pull request merged**